app/Criteria/ForOrganizationDiscountSearch.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace App\Criteria;
+
+use App\Enums\ConditionType;
+use Heseya\Searchable\Criteria\Criterion;
+use Illuminate\Database\Eloquent\Builder;
+
+class ForOrganizationDiscountSearch extends Criterion
+{
+    public function query(Builder $query): Builder
+    {
+        return $query->whereHas('conditionGroups', function (Builder $query): void {
+            $query->whereHas('conditions', function (Builder $query): void {
+                $query->where(function (Builder $query): void {
+                    $query->where('type', ConditionType::USER_IN_ORGANIZATION->value)
+                        ->where('value->is_allow_list', true)
+                        ->where('value', 'LIKE', "%{$this->value}%");
+                })->orWhere(function (Builder $query): void {
```

Comment:
What if a value is both in allow list and block list?

Reply:
If such conditions are in separate groups, it will work. The problem will be if both conditions end up in the same group